image/preseed/preseed_uc20_test.go

```diff
@@ -27,13 +27,91 @@ import (
 
 	. "gopkg.in/check.v1"
 
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/asserts/assertstest"
+	"github.com/snapcore/snapd/asserts/signtool"
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/image/preseed"
 	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/seed"
+	"github.com/snapcore/snapd/snap"
 	"github.com/snapcore/snapd/testutil"
 )
 
+type fakeKeyMgr struct {
+	key asserts.PrivateKey
+}
+
+func (f *fakeKeyMgr) Put(privKey asserts.PrivateKey) error                  { return nil }
+func (f *fakeKeyMgr) Get(keyID string) (asserts.PrivateKey, error)          { return f.key, nil }
+func (f *fakeKeyMgr) Delete(keyID string) error                             { return nil }
+func (f *fakeKeyMgr) GetByName(keyNname string) (asserts.PrivateKey, error) { return f.key, nil }
+func (f *fakeKeyMgr) Export(keyName string) ([]byte, error)                 { return nil, nil }
+func (f *fakeKeyMgr) List() ([]asserts.ExternalKeyInfo, error)              { return nil, nil }
+func (f *fakeKeyMgr) DeleteByName(keyName string) error                     { return nil }
+
+func mockUC20Model() *asserts.Model {
+	headers := map[string]interface{}{
+		"type":         "model",
+		"authority-id": "my-brand",
+		"series":       "16",
+		"brand-id":     "my-brand",
+		"model":        "my-model-uc20",
+		"display-name": "My Model",
+		"architecture": "amd64",
+		"base":         "core20",
+		"grade":        "dangerous",
+		"timestamp":    "2019-11-01T08:00:00+00:00",
+		"snaps": []interface{}{
+			map[string]interface{}{
+				"name": "pc-kernel",
+				"id":   "pckernelidididididididididididid",
+				"type": "kernel",
+			},
+			map[string]interface{}{
+				"name": "pc",
+				"id":   "pcididididididididididididididid",
+				"type": "gadget",
+			},
+		},
+	}
+	return assertstest.FakeAssertion(headers, nil).(*asserts.Model)
+}
+
 func (s *preseedSuite) TestRunPreseedUC20Happy(c *C) {
+	restoreSeedOpen := preseed.MockSeedOpen(func(rootDir, label string) (seed.Seed, error) {
+		return &FakeSeed{
+			AssertsModel: mockUC20Model(),
+			UsesSnapd:    true,
+			Essential: []*seed.Snap{{
+				Path: "/some/path/snapd.snap",
+				SideInfo: &snap.SideInfo{
+					RealName: "snapd",
+					SnapID:   "snapdidididididididididididididd",
+					Revision: snap.R("1")}},
+			},
+			SnapsForMode: map[string][]*seed.Snap{
+				"run": {{
+					Path: "/some/path/foo.snap",
+					SideInfo: &snap.SideInfo{
+						RealName: "foo"},
+				}}},
+		}, nil
+	})
+	defer restoreSeedOpen()
+
+	restoreSaveAssertion := preseed.MockSaveAssertion(func(*asserts.Database, asserts.Assertion, *asserts.Model) error {
```

Comment:
this is mocking too much here,  this way we don't need to deal with the account-key that corresponds to the key we create below, but it's important that the real code saves that account-key as I mentioned elsewhere